### PR TITLE
Step2

### PR DESCRIPTION
**Pull request by welliam**

### CODE REVIEW

File: src/server.py [hunk 1]
```diff
@@ -26,14 +38,74 @@ def response_ok():
...
+        raise HTTPException(HTTP_SERVER_ERROR, 'Method is not GET')
+    if http_version != 'HTTP/1.1':
+        raise HTTPException(HTTP_SERVER_ERROR, 'HTTP version is not 1.1')
+    if 'host' not in headers:
+        raise HTTPException(HTTP_CLIENT_ERROR, 'Host not in header')
+
+
+def parse_request(request):
+    head, body = split_head(request)
+    head_lines = list(filter(lambda x: x, head.split('\r\n')))
```
Comment (cewing) [inline src/server.py]: Can you tell me how this is not exactly the same as
```python
head_lines = [x for x in head.split("\r\n") if x]
```
?

File: src/server.py [hunk 2]
```diff
@@ -26,14 +38,74 @@ def response_ok():
...
+    except ValueError:
+        raise HTTPException(
+            HTTP_CLIENT_ERROR,
+            'Header line has no colon'
+        )
+
+
+def verify_head(method, http_version, headers):
+    if method != 'GET':
+        raise HTTPException(HTTP_SERVER_ERROR, 'Method is not GET')
```
Comment (cewing) [inline src/server.py]: Is this really a server error?  It's not the server's fault the client sent a request with the wrong method.  Is there a 400-level error that better fits this case?

File: src/server.py [hunk 3]
```diff
@@ -26,14 +38,74 @@ def response_ok():
...
+    if http_version != 'HTTP/1.1':
+        raise HTTPException(HTTP_SERVER_ERROR, 'HTTP version is not 1.1')
```
Comment (cewing) [inline src/server.py]: see above

File: src/server.py (hunk 2 above)
Comment (welliam) [inline src/server.py]: I tried POSTing to a website that didn't accept it and got back a 404. Is that what we're supposed to report?

**Pull request merged**